R/decostand.R

```diff
@@ -260,3 +277,348 @@
         x[abs(x) < sqrt(.Machine$double.eps)] <- 0
     x
 }
+
+
+
+
+
+# .OptSpace : an algorithm for matrix reconstruction from a partially revealed set
+# This function has been adapted from the original source code in the ROptSpace R package
+# (version 0.2.3) by
+# Raghunandan H. Keshavan, Andrea Montanari, Sewoong Oh (2010).
+# See the ROptSpace::OptSpace for more information.
+# Let's assume an ideal matrix \eqn{M} with \eqn{(m\times n)} entries with rank \eqn{r} and
+# we are given a partially observed matrix \eqn{M\_E} which contains many missing entries.
+# Matrix reconstruction - or completion - is the task of filling in such entries.
+# OptSpace is an efficient algorithm that reconstructs \eqn{M} from \eqn{|E|=O(rn)}
+# observed elements with relative root mean square error (RMSE)
+# \deqn{RMSE \le C(\alpha)\sqrt{nr/|E|}}
+#
+# @param A an \eqn{(n\times m)} matrix whose missing entries should be flaged as NA.
+# @param ropt \code{NA} to guess the rank, or a positive integer as a pre-defined rank.
+# @param niter maximum number of iterations allowed.
+# @param tol stopping criterion for reconstruction in Frobenius norm.
+# @param showprogress a logical value; \code{TRUE} to show progress, \code{FALSE} otherwise.
+#
+# @return a named list containing
+# \describe{
+# \item{X}{an \eqn{(n \times r)} matrix as left singular vectors.}
+# \item{S}{an \eqn{(r \times r)} matrix as singular values.}
+# \item{Y}{an \eqn{(m \times r)} matrix as right singular vectors.}
+# \item{dist}{a vector containing reconstruction errors at each successive iteration.}
+# }
+# @references
+# Raghunandan H. Keshavan, Andrea Montanari, Sewoong Oh (2010).
+# Matrix Completion From a Few Entries.
+# IEEE Transactions on Information Theory 56(6):2980--2998.
+#
+.OptSpace <- function(A, ropt=NA, niter=50, tol=1e-6, showprogress=FALSE){
```

Comment:
Other vegan functions uses trace = FALSE instead of showprogress. Wouldn't hurt here, although not crucial. These functions also usually cat() tracing instead of message().

Vegan usually has opening { on a new line in function definition (again: not crucial).

R-ext manual says that stop() (and warning()) messages should be seen as a part of a sentence and should not have period / full stop (.) at the end.

Reply:
message changed to cat

openings updated

periods removed